coderd/database/dbauthz/dbauthz.go

```diff
@@ -1791,6 +1791,10 @@ func (q *querier) GetProvisionerJobByID(ctx context.Context, id uuid.UUID) (data
 	return job, nil
 }
 
+func (q *querier) GetProvisionerJobTimingsByJobID(ctx context.Context, jobID uuid.UUID) ([]database.ProvisionerJobTiming, error) {
+	return q.db.GetProvisionerJobTimingsByJobID(ctx, jobID)
```

Comment:
@BrunoQuaresma I missed this yesterday; we need to ensure the user can read the workspace build in order to read the timings, otherwise we might leak some information.

Reply:
+1

Reply:
I think you can take a look at `GetProvisionerLogsAfterID`. As you can see, it is assumed that if you can fetch Job from the database, you can also read _Logs_

Reply:
@mtojek even better, ty

Reply:
https://github.com/coder/coder/pull/14697